.github/workflows/lint.yml

```diff
@@ -15,8 +19,46 @@ jobs:
     timeout-minutes: 10
 
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
       - uses: actions/setup-python@v4
         with:
           python-version: "3.x"
       - uses: pre-commit/action@v3.0.0
+
+  patchcheck:
+    if: github.event_name == 'pull_request'
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          # Check out the original HEAD of the PR branch.
+          # Note that this value is cached by GitHub, so further commits to the
+          # branch will not be reflected in this checkout.
+          ref: ${{ github.event.pull_request.head.sha }}
+          show-progress: false
+      # Adapted from https://github.com/actions/checkout/issues/520#issuecomment-1167205721
+      - name: Fetch merge-base
+        run: >
+          git fetch
+          origin
+          +${{ github.event.pull_request.head.sha }}:origin/${{ github.event.pull_request.head.ref }}
+          --depth=$(( ${{ github.event.pull_request.commits }} + 1 ))
+          --no-tags
+          --prune
+          --no-recurse-submodules
+      - name: Fetch base branch
+        run: >
+          git fetch
+          origin
+          +${{ github.event.pull_request.base.sha }}:origin/${{ github.event.pull_request.base.ref }}
+          --depth=5
```

Comment:
Why 5?

Reply:
As the azure job uses a depth of 5 which seemed to work and is simpler than doing the full refspec dance